frontend/src/components/InstanceForm/common.ts

```diff
@@ -80,14 +81,50 @@ export const extractBasicInfo = (instance: Instance | undefined): BasicInfo => {
 };
 
 export const wrapEditDataSource = (ds: DataSource | undefined) => {
-  return {
-    ...cloneDeep(ds ?? emptyDataSource()),
+  // Deep clone the data source to avoid reference issues
+  const cloned = cloneDeep(ds ?? emptyDataSource());
+  
+  // Create a plain object from potentially proxied extraConnectionParameters
+  const createPlainParamsObject = (obj: any) => {
+    if (!obj) return {};
+    
+    const result: Record<string, string> = {};
+    
+    // This handles both plain objects and Proxy objects
+    try {
+      // Get keys and copy each property
+      const keys = Object.keys(obj);
+      keys.forEach(key => {
+        result[key] = obj[key];
+      });
+      
+      // Also try using Object.entries as a backup
+      Object.entries(obj).forEach(([key, value]) => {
+        result[key] = value as string;
+      });
+    } catch {
+      // Silent catch - if we can't access properties, return empty object
+    }
```

Comment:
The `cloned.extraConnectionParameters` should already be cloned, seems we don't need the `createPlainParamsObject`